model_v4.py

```diff
@@ -0,0 +1,382 @@
+        self.inception_5b = self.create_interception_block(self.inception_5a, addition_name='5b',
+                                                           conv1_5x5=None, conv2_5x5=None,
+                                                           conv1_3x3=96, conv2_3x3=384,
+                                                           conv_filters=256, pool_filters=96,
+                                                           pool_type='max')
+
+        # Common last layers
+        avg = GlobalAveragePooling2D(name='avg_pool')(self.inception_5b)
+        self.flatten_layer = Flatten()(avg)
+        self.dense_layer = Dense(units=128, kernel_regularizer=l2(0.01))(self.flatten_layer)
+        self.out_layer = Lambda(lambda x: l2_normalize(x, axis=1))(self.dense_layer)
+
+        model = Model(input_size, self.out_layer)
+        model.summary()
+
+        self.model = model
+
+    @staticmethod
+    def get_all_weights(root_weights_path):
+        """
+        Create dict filename: weights_values
+        :param root_weights_path: path to folder with weights
+        :return: dictionary {filename: weights_values}
+        """
+        full_list = os.listdir(root_weights_path)
+        # TODO: regular expresions
+        layer_name_selector = lambda filename: filename.split('.')[0]
+        weight_loader = lambda filename: np.loadtxt(os.path.join(root_weights_path, filename), delimiter=',',
+                                                                                               dtype=np.float32)
+        model_weights_by_layers = {layer_name_selector(filename): weight_loader(filename) for filename in full_list}
+        return model_weights_by_layers
+
+    def load_weights(self, root_weights_path):
+        """
+        Load weights from filer to model
+        :param root_weights_path:  path to folder with weights
+        :return: None
+        """
+        model_weights_by_layers = self.get_all_weights(root_weights_path)
+        grouped_weights = {}
+        for x in model_weights_by_layers:
+            main_name = x[:-2]
+            if main_name in grouped_weights:
+                grouped_weights[main_name].append(x)
+            else:
+                grouped_weights[main_name] = [x]
+
+        for layer_name, files_names in grouped_weights.items():
+           files_names = sorted(files_names) #bw/bmvw
+           order_set = self.order_w_name_idx if len(files_names) == 4 else self.order_b_name_idx
+           files_names = [x for _, x in sorted(zip(order_set, files_names), key=lambda pair: pair[0])]
+
+           try:
+                layer = self.model.get_layer(name=layer_name)
+                src_weights = layer.get_weights()
+                weights_list = [model_weights_by_layers[filename] for filename in files_names]
+                weights_list[0] = weights_list[0].reshape(src_weights[0].shape)
+                layer.set_weights(weights_list)
```

Comment:
It will not work as expected, because weights are stored as NCHW tensor (or CoCiHW in terms of TensorFlow) with shape specified in the task Appendix. Therefore, you have to firstly reshape your tensor and physically transpose it to HWCiCo after that.

Reply:
I don't see transposing here. You just need to do the following:
```
conv_weights = np.reshape(conv_weights , conv_shape[name])
conv_weights  = np.transpose(conv_weights , (2, 3, 1, 0))
```